tqdm.py

```diff
@@ -0,0 +1,64 @@
+import time
+from inspect import isgenerator
+
+PROGRESS_BAR_LEN = 20
+
+
+class Tqdm(object):
+    """
+    Iterable that prints status of other iterable objects
+    """
+
+    def __init__(self, iterable):
+        """
+        :param iterable: Iterable object (has the method __iter__) to print his status
+        """
+        self.iterable = iterable if not isgenerator(iterable) else list(iterable)
```

Comment:
Missing the point when calling `list(iterable)`, init is going to take all the time.